### PR TITLE
feat(db-features) Implement 'database_features' in addons to enable force-ssl and other features through the provider

### LINKED ISSUE #90: Add ability to set features on a database addon ie. ('force-ssl', 'public-availability')

### PR DESCRIPTION
**Pull request by Soulou**
Fixes #90

### CODE REVIEW

**Review by curzolapierre — APPROVED** [review]
LGTM

File: scalingo/resource_scalingo_addon.go [hunk 1]
```diff
@@ -219,3 +328,25 @@ func waitUntilProvisionned(ctx context.Context, client *scalingo.Client, addon s
...
+		if err != nil {
+			return fmt.Errorf("fail to refresh database metadata: %w", err)
+		}
+		for _, f := range db.Features {
+			if f.Name == feature && f.Status != scalingo.DatabaseFeatureStatusPending {
+				switch f.Status {
+				case scalingo.DatabaseFeatureStatusActivated:
+					return nil
+				case scalingo.DatabaseFeatureStatusFailed:
+					return nil
```
Comment (EtienneM) [inline scalingo/resource_scalingo_addon.go:346]: question: shouldn't we return an error here?

File: scalingo/resource_scalingo_addon.go [hunk 2]
```diff
@@ -111,6 +131,28 @@ func resourceAddonRead(ctx context.Context, d *schema.ResourceData, meta interfa
...
 
+	providers, err := client.AddonProvidersList(ctx)
+	if err != nil {
+		return diag.Errorf("fail to list addon providers: %v", err)
+	}
+	addonProvider := keepIf(providers, func(p *scalingo.AddonProvider) bool {
+		return p.ID == addon.AddonProvider.ID
+	})[0]
+	features := []string{}
+	if addonProvider.Category.Name == "Database" {
```
Comment (EtienneM) [inline scalingo/resource_scalingo_addon.go:142]: In staging, this category is named `Databases` (https://admin.osc-st-fr1.st-sc.fr/platform_data/addons). Isn't there a more reliable way to get the databases addons? :/

Comment (Soulou) [reply scalingo/resource_scalingo_addon.go:142]: Not really :(

File: scalingo/resource_scalingo_addon.go (hunk 1 above)
Comment (Soulou) [reply scalingo/resource_scalingo_addon.go]: ACK

File: scalingo/resource_scalingo_addon.go (hunk 2 above)
Comment (Soulou) [reply scalingo/resource_scalingo_addon.go]: ACK

**Review by EtienneM — APPROVED** [review]

**Pull request merged**